### DISCUSSION TITLE
Production build not compiling/consuming the custom server.js file

### DISCUSSION THREAD

**Discussion by kandukurivarun** [vercel/next.js#51892] [Help]
### Summary

I have setup a custom server.js file based on our needs and when I do `npm run dev`(internally runs `node server.js`), everything works perfect. But when I do the `npm run build`, the `.next` folder is not containing the custom server.js file that I placed in the root of the application. `.next/standalone` folder contains the default server.js provided by next and not the one that I customized. Am I missing anything? How do I let my `next build` know to also compile and consume the custom server.js file?

**Comment by icyJoseph** [comment]
Yeah, I am not sure this is gonna fly, because of the standalone mode: https://github.com/vercel/next.js/discussions/34599#discussion-3888962

Normally, without standalone, all you gotta do is reuse your `server.js`:

```json
{
  "scripts": {
    "dev": "node server.js",
    "build": "next build",
    "start": "NODE_ENV=production node server.js"
  }
}
```

Something like that. However, since standalone outputs its own, then you might have problems. 

This means that, without standalone mode, you gotta copy the server.js along with the `.next` folder.

> server.js doesn't go through babel or webpack. Make sure the syntax and sources this file requires are compatible with the current node version you are running.

This thread has people covering this very topic: https://github.com/vercel/next.js/discussions/34599#discussion-3888962

**Reply by kandukurivarun** [reply]
@icyJoseph Thank you for your response. I just tried this but It looks like I need to bundle my `server.js` as well as it has references to the folders that are not gonna exist in the Production build. I tried bundling the `server.js` using `webpack-cli` and that is also not working as expected. Trying to find a way to lets me use my custom server.js. I checked on the thread you mentioned but none of the solutions seemed to work.

**Reply by icyJoseph** [reply]
No, I see because the custom server is mostly all about spinning up the app. 

This is almost a custom-custom server. Not sure how to proceed - 

I saw a person that was doing this programmatically, probably trying to do the same thing as you do. https://github.com/vercel/next.js/discussions/51515
